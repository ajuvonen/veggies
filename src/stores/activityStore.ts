import {computed, ref} from 'vue';
import {defineStore, storeToRefs} from 'pinia';
import {debounceFilter, useIntervalFn, useNow, useStorage} from '@vueuse/core';
import {DateTime} from 'luxon';
import {
  countBy,
  difference,
  entries,
  intersection,
  map,
  pipe,
  prop,
  sortBy,
  take,
  takeWhile,
} from 'remeda';
import {
  Category,
  type Favorites,
  type Challenge,
  type Week,
  type Achievements,
  AchievementLevel,
} from '@/utils/types';
import {
  achievementLevelHelper,
  dateParser,
  dateReplacer,
  getCategoryForVeggie,
  getRandomVeggie,
} from '@/utils/helpers';
import {useAppStateStore} from '@/stores/appStateStore';
import {
  BEANS,
  BOTANICAL_BERRIES,
  FRUITS,
  GRAINS,
  LEAFIES,
  MUSHROOMS,
  NUTS,
  RED_VEGGIES,
  ROOTS,
  VEGETABLES,
} from '@/utils/constants';

export const useActivityStore = defineStore('activity', () => {
  const {settings} = storeToRefs(useAppStateStore());
  const currentDate = ref(DateTime.now());
  useIntervalFn(() => {
    const now = DateTime.now();
    if (!currentDate.value.hasSame(now, 'day')) {
      currentDate.value = now;
    }
  }, 2000);

  // State refs
  const startDate = useStorage<DateTime | null>('veggies-start-date', null, localStorage, {
    mergeDefaults: true,
    serializer: {
      read: (v) => (v ? DateTime.fromISO(v.split('T')[0]) : null),
      write: (v) => v?.toISODate() ?? '',
    },
  });

  const weeks = useStorage<Week[]>('veggies-weeks', [], localStorage, {
    mergeDefaults: true,
    eventFilter: debounceFilter(2000),
    serializer: {
      read: (v) => (v ? JSON.parse(v, dateParser) : null),
      write: (v) => JSON.stringify(v, dateReplacer),
    },
  });

  const challenges = useStorage<Challenge[]>('veggies-challenges', [], localStorage, {
    mergeDefaults: true,
    serializer: {
      read: (v) => (v ? JSON.parse(v, dateParser) : null),
      write: (v) => JSON.stringify(v, dateReplacer),
    },
  });

  // Computed getters
  const currentWeekStart = computed(() => currentDate.value.startOf('week'));

  const getWeekStarts = computed(() => {
    if (!startDate.value) return [currentWeekStart.value];
    const totalWeeks = Math.ceil(currentDate.value.diff(startDate.value, 'week').weeks);
    return Array.from({length: totalWeeks}, (_, weekIndex) =>
      currentWeekStart.value.minus({weeks: weekIndex}),
    );
  });

  const hotStreak = computed(() => {
    let over30Weeks = takeWhile(
      getWeekStarts.value,
      (weekStart, weekIndex) => weekIndex === 0 || veggiesForWeek.value(weekStart).length >= 30,
    ).length;
    if (currentVeggies.value.length < 30) {
      over30Weeks--;
    }
    return over30Weeks;
  });

  const allVeggies = computed(() => weeks.value.flatMap(prop('veggies')));

  const uniqueVeggies = computed(() => [...new Set(allVeggies.value)]);

  const over30Veggies = computed(
    () => weeks.value.filter(({veggies}) => veggies.length >= 30).length,
  );

  const atMostVeggies = computed(() =>
    Math.max(...weeks.value.map(({veggies}) => veggies.length), 0),
  );

  const completedChallenges = computed(
    () =>
      challenges.value.filter(({startDate, veggie}) =>
        veggiesForWeek.value(startDate).includes(veggie),
      ).length,
  );

  const veggiesForWeek = computed(
    () => (weekStart: DateTime) =>
      weeks.value.find(({startDate}) => startDate.equals(weekStart))?.veggies ?? [],
  );

  const currentVeggies = computed({
    get: () => veggiesForWeek.value(currentWeekStart.value),
    set: (veggies: string[]) => setVeggiesForWeek(veggies),
  });

  const currentChallenge = computed(
    () => challenges.value.find(({startDate}) => startDate.equals(currentWeekStart.value))?.veggie,
  );

  const suggestions = computed(() =>
    pipe(
      allVeggies.value.filter((veggie) => !currentVeggies.value.includes(veggie)),
      countBy((veggie) => veggie),
      entries(),
      sortBy([prop(1), 'desc']),
      take(settings.value.suggestionCount),
      map(prop(0)),
    ),
  );

  const favorites = computed(() =>
    Object.values(Category).reduce(
      (acc, category) => ({
        ...acc,
        [category]: pipe(
          allVeggies.value.filter((veggie) => getCategoryForVeggie(veggie) === category),
          countBy((veggie) => veggie),
          entries(),
          sortBy([prop(1), 'desc']),
          take(6),
        ),
      }),
      {} as Favorites,
    ),
  );

  const weeklyAchievements = computed(() => (veggies: string[] = currentVeggies.value) => ({
    allOnRed: achievementLevelHelper(
      [[10, AchievementLevel.Gold]],
      intersection(RED_VEGGIES, veggies).length,
    ),
    botanicalBerries: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(BOTANICAL_BERRIES, veggies).length,
    ),
    goNuts: achievementLevelHelper(
      [[5, AchievementLevel.Gold]],
      intersection(NUTS, veggies).length,
    ),
    thirtyVeggies: achievementLevelHelper(
      [
        [40, AchievementLevel.Platinum],
        [30, AchievementLevel.Gold],
      ],
      veggies.length,
    ),
  }));

  const achievements = computed<Achievements>(() => ({
    challengeAccepted: achievementLevelHelper(
      [
        [20, AchievementLevel.Gold],
        [10, AchievementLevel.Silver],
        [5, AchievementLevel.Bronze],
      ],
      completedChallenges.value,
    ),
    committed: achievementLevelHelper(
      [
        [52, AchievementLevel.Gold],
        [26, AchievementLevel.Silver],
        [12, AchievementLevel.Bronze],
      ],
      weeks.value.length,
    ),
    completionist: achievementLevelHelper(
      [
        [150, AchievementLevel.Gold],
        [80, AchievementLevel.Silver],
        [40, AchievementLevel.Bronze],
      ],
      uniqueVeggies.value.length,
    ),
    experimenterBean: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(BEANS, uniqueVeggies.value).length,
    ),
    experimenterFruit: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(FRUITS, uniqueVeggies.value).length,
    ),
    experimenterGrain: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(GRAINS, uniqueVeggies.value).length,
    ),
    experimenterLeafy: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(LEAFIES, uniqueVeggies.value).length,
    ),
    experimenterMushroom: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(MUSHROOMS, uniqueVeggies.value).length,
    ),
    experimenterRoot: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(ROOTS, uniqueVeggies.value).length,
    ),
    experimenterVegetable: achievementLevelHelper(
      [[15, AchievementLevel.Gold]],
      intersection(VEGETABLES, uniqueVeggies.value).length,
    ),
    hotStreak: achievementLevelHelper(
      [
        [20, AchievementLevel.Gold],
        [10, AchievementLevel.Silver],
        [5, AchievementLevel.Bronze],
      ],
      hotStreak.value,
    ),
    thousandsEven:
      allVeggies.value.length >= 1000 && Math.floor(allVeggies.value.length / 1000) % 2 === 0
        ? AchievementLevel.Platinum
        : AchievementLevel.NoAchievement,
    thousandsOdd:
      allVeggies.value.length >= 1000 && Math.floor(allVeggies.value.length / 1000) % 2 === 1
        ? AchievementLevel.Platinum
        : AchievementLevel.NoAchievement,
    ...weeklyAchievements.value(),
  }));

  // Actions
  const toggleVeggie = (targetVeggie: string) =>
    toggleVeggieForWeek(targetVeggie, currentWeekStart.value);

  const toggleVeggieForWeek = (targetVeggie: string, weekStart: DateTime) => {
    const targetWeek = weeks.value.find(({startDate}) => startDate.equals(weekStart));
    if (!targetWeek) {
      weeks.value = [
        ...weeks.value,
        {
          startDate: weekStart,
          veggies: [targetVeggie],
        },
      ];
      challenges.value = [
        ...challenges.value,
        {
          startDate: weekStart,
          veggie: getRandomVeggie(),
        },
      ];
    } else if (!targetWeek.veggies.includes(targetVeggie)) {
      targetWeek.veggies = [...targetWeek.veggies, targetVeggie];
    } else {
      targetWeek.veggies = difference(targetWeek.veggies, [targetVeggie]);
    }
  };

  const setVeggiesForWeek = (veggies: string[], weekStart: DateTime = currentWeekStart.value) => {
    const targetWeek = weeks.value.find(({startDate}) => startDate.equals(weekStart));
    if (!targetWeek) {
      weeks.value = [
        ...weeks.value,
        {
          startDate: weekStart,
          veggies,
        },
      ];
      challenges.value = [
        ...challenges.value,
        {
          startDate: weekStart,
          veggie: getRandomVeggie(),
        },
      ];
    } else {
      targetWeek.veggies = veggies;
    }
  };

  const $reset = () => {
    startDate.value = null;
    weeks.value = [];
    challenges.value = [];
  };

  return {
    achievements,
    allVeggies,
    atMostVeggies,
    challenges,
    completedChallenges,
    currentChallenge,
    currentVeggies,
    favorites,
    getWeekStarts,
    hotStreak,
    over30Veggies,
    startDate,
    suggestions,
    uniqueVeggies,
    veggiesForWeek,
    weeklyAchievements,
    weeks,
    setVeggiesForWeek,
    toggleVeggie,
    toggleVeggieForWeek,
    $reset,
  };
});
