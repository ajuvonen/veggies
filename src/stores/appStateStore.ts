import {ref, watchEffect} from 'vue';
import {defineStore, storeToRefs} from 'pinia';
import {useStorage} from '@vueuse/core';
import type {Settings} from '@/utils/types';
import {useAchievements} from '@/hooks/achievements';
import {useActivityStore} from '@/stores/activityStore';

type Message = {
  id: string;
  text: string;
};

export const useAppStateStore = defineStore('appState', () => {
  const {advance, achievements, reset} = useAchievements();
  const {uniqueVeggies} = storeToRefs(useActivityStore());

  watchEffect(() => {
    advance(uniqueVeggies.value, 0);
  });

  // State refs
  const settings = useStorage<Settings>(
    'veggies-settings',
    {
      locale: 'en',
    },
    localStorage,
    {
      mergeDefaults: true,
    },
  );

  const messages = ref<Message[]>([]);

  const addToastMessage = (text: string) => {
    messages.value.push({
      id: crypto.randomUUID(),
      text,
    });
  };

  const removeToastMessage = (targetId: string) => {
    messages.value = messages.value.filter(({id}) => id !== targetId);
  };

  const $reset = () => {
    messages.value = [];
    settings.value = {
      locale: 'en',
    };
    reset();
  };

  return {
    messages,
    settings,
    achievements,
    addToastMessage,
    removeToastMessage,
    $reset,
  };
});
