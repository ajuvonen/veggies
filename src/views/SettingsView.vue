<script setup lang="ts">
import {ref} from 'vue';
import {useRouter} from 'vue-router';
import {useActivityStore} from '@/stores/activityStore';
import {useAppStateStore} from '@/stores/appStateStore';
import LocaleChanger from '@/components/LocaleChanger.vue';
import ModalDialog from '@/components/ModalDialog.vue';
import QAComponent from '@/components/QAComponent.vue';
import BuildTime from '@/components/BuildTime.vue';

const router = useRouter();

const {$reset: activityReset} = useActivityStore();
const {$reset: appStateReset} = useAppStateStore();

const resetDialogOpen = ref(false);

const reset = () => {
  activityReset();
  appStateReset();
  router.push({name: 'home'});
};
</script>

<template>
  <h1 class="sr-only">{{ $t('views.settings') }}</h1>
  <LocaleChanger />
  <QAComponent />
  <ContentElement
    :title="$t('settings.reset.title')"
    :labelAttrs="{for: 'reset-button'}"
    labelTag="label"
  >
    <ButtonComponent
      id="reset-button"
      variant="danger"
      icon="trashCan"
      data-test-id="reset-button"
      class="self-end"
      @click="resetDialogOpen = true"
      >{{ $t('settings.reset.button') }}</ButtonComponent
    >
  </ContentElement>
  <BuildTime />
  <ModalDialog v-model="resetDialogOpen" :title="$t('settings.reset.title')">
    <template #content>
      <p>{{ $t('settings.reset.text') }}</p>
    </template>
    <template #buttons>
      <ButtonComponent data-test-id="cancel-button" @click="resetDialogOpen = false">{{
        $t('general.cancel')
      }}</ButtonComponent>
      <ButtonComponent
        data-test-id="confirm-button"
        variant="danger"
        icon="trashCan"
        @click="reset()"
        >{{ $t('settings.reset.button') }}</ButtonComponent
      >
    </template>
  </ModalDialog>
</template>
