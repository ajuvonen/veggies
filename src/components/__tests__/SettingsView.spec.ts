import {computed, nextTick} from 'vue';
import {describe, it, expect, beforeEach, vi, afterEach} from 'vitest';
import {mount} from '@vue/test-utils';
import {useAppStateStore} from '@/stores/appStateStore';
import {useActivityStore} from '@/stores/activityStore';
import SettingsView from '@/views/SettingsView.vue';
import DialogStub from './DialogStub.vue';

const mocks = vi.hoisted(() => ({
  usePreferredReducedMotion: vi.fn(() => computed(() => 'no-preference')),
}));

vi.mock('@vueuse/core', async () => {
  const actual = await vi.importActual('@vueuse/core');
  return {
    ...actual,
    usePreferredReducedMotion: mocks.usePreferredReducedMotion,
  };
});

const mounter = () =>
  mount(SettingsView, {
    global: {
      stubs: {
        Dialog: DialogStub,
        DialogPanel: {
          template: '<div><slot /></div>',
        },
        DialogTitle: true,
      },
    },
  });

describe('SettingsView', () => {
  let activityStore: ReturnType<typeof useActivityStore>;
  let appStateStore: ReturnType<typeof useAppStateStore>;

  beforeEach(() => {
    activityStore = useActivityStore();
    appStateStore = useAppStateStore();
  });

  afterEach(() => {
    vi.restoreAllMocks();
  });

  it('renders', () => {
    const wrapper = mount(SettingsView, {
      global: {
        stubs: {
          BuildTime: true,
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('toggles animations', async () => {
    const wrapper = mounter();
    expect(appStateStore.settings.showAnimations).toBe(true);
    const toggle = wrapper.findByTestId('show-animations-button');
    expect(toggle.attributes('disabled')).toBe(undefined);
    expect(toggle.attributes('data-headlessui-state')).toBe('checked');
    await toggle.trigger('click');
    expect(appStateStore.settings.showAnimations).toBe(false);
    expect(toggle.attributes('data-headlessui-state')).not.toBe('checked');
  });

  it('prevents animation toggle if all animations are disabled', () => {
    mocks.usePreferredReducedMotion.mockImplementation(() => computed(() => 'reduce'));
    const wrapper = mounter();
    const toggle = wrapper.findByTestId('show-animations-button');
    expect(appStateStore.settings.showAnimations).toBe(true);
    expect(toggle.attributes('disabled')).not.toBe(undefined);
    expect(toggle.attributes('data-headlessui-state')).not.toBe('checked');
  });

  it('resets the app', async () => {
    const wrapper = mounter();
    await wrapper.findByTestId('reset-button').trigger('click');
    await wrapper.findByTestId('confirm-button').trigger('click');
    expect(appStateStore.$reset).toBeCalledTimes(1);
    expect(activityStore.$reset).toBeCalledTimes(1);
  });

  it('cancels reset', async () => {
    const wrapper = mounter();
    await wrapper.findByTestId('reset-button').trigger('click');
    await wrapper.findByTestId('cancel-button').trigger('click');
    expect(appStateStore.$reset).toBeCalledTimes(0);
    expect(activityStore.$reset).toBeCalledTimes(0);
  });

  it('starts file download', async () => {
    const wrapper = mounter();
    const link = document.createElement('a');
    link.click = vi.fn();
    vi.stubGlobal('URL', {
      createObjectURL: vi.fn(() => 'https://eatyourveggies.app/'),
      revokeObjectURL: vi.fn(),
    });
    const createElementSpy = vi.spyOn(document, 'createElement').mockImplementation(() => link);

    wrapper.findByTestId('export-button').trigger('click');
    await nextTick();
    try {
      expect(link.href).toBe('https://eatyourveggies.app/');
      expect(link.click).toHaveBeenCalledTimes(1);
    } finally {
      vi.unstubAllGlobals();
      createElementSpy.mockRestore();
    }
  });
});
