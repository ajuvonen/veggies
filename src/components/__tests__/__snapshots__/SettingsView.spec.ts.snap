// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SettingsView > renders 1`] = `
"<h1 class="sr-only">Settings</h1>
<div id="headlessui-radiogroup-v-0" role="radiogroup">
  <div class="flex-container flex-col"><label id="headlessui-label-v-1" class="label-like">Language</label>
    <div id="headlessui-radiogroup-option-v-2" role="radio" aria-checked="true" tabindex="0" data-headlessui-state="checked" data-test-id="locale-button-en" class="button-like bg-sky-500 hover:bg-sky-600"><svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2M12,7A5,5 0 0,0 7,12A5,5 0 0,0 12,17A5,5 0 0,0 17,12A5,5 0 0,0 12,7Z" class="transition-all"></path>
      </svg><span id="headlessui-label-v-3">English</span></div>
    <div id="headlessui-radiogroup-option-v-4" role="radio" aria-checked="false" tabindex="-1" data-headlessui-state="" data-test-id="locale-button-fi" class="button-like bg-sky-500 hover:bg-sky-600"><svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4A8,8 0 0,1 20,12A8,8 0 0,1 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z" class="transition-all"></path>
      </svg><span id="headlessui-label-v-5">Suomi</span></div>
  </div>
</div>
<div class="flex-container flex-col min-h-0"><label class="label-like">Common Questions</label>
  <div class="flex-container flex-col has-scroll"><button data-v-24e38f3d="" class="button-like button--primary w-full justify-between" id="headlessui-disclosure-button-v-6" aria-expanded="false" data-headlessui-state="" data-test-id="qa-button-appPurpose">
      <!--v-if-->What's this app for? <svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z" class="transition-all"></path>
      </svg>
    </button>
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform h-0 opacity-0" entertoclass="transform h-full opacity-100" leaveactiveclass="transition duration-75 ease-out" leavefromclass="transform h-full opacity-100" leavetoclass="transform h-0 opacity-0" appear="false" persisted="false" css="true">
      <!---->
    </transition-stub><button data-v-24e38f3d="" class="button-like button--primary w-full justify-between" id="headlessui-disclosure-button-v-8" aria-expanded="false" data-headlessui-state="" data-test-id="qa-button-studyDetails">
      <!--v-if-->What's the science? <svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z" class="transition-all"></path>
      </svg>
    </button>
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform h-0 opacity-0" entertoclass="transform h-full opacity-100" leaveactiveclass="transition duration-75 ease-out" leavefromclass="transform h-full opacity-100" leavetoclass="transform h-0 opacity-0" appear="false" persisted="false" css="true">
      <!---->
    </transition-stub><button data-v-24e38f3d="" class="button-like button--primary w-full justify-between" id="headlessui-disclosure-button-v-10" aria-expanded="false" data-headlessui-state="" data-test-id="qa-button-whatCounts">
      <!--v-if-->What counts as a portion? <svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z" class="transition-all"></path>
      </svg>
    </button>
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform h-0 opacity-0" entertoclass="transform h-full opacity-100" leaveactiveclass="transition duration-75 ease-out" leavefromclass="transform h-full opacity-100" leavetoclass="transform h-0 opacity-0" appear="false" persisted="false" css="true">
      <!---->
    </transition-stub><button data-v-24e38f3d="" class="button-like button--primary w-full justify-between" id="headlessui-disclosure-button-v-12" aria-expanded="false" data-headlessui-state="" data-test-id="qa-button-categories">
      <!--v-if-->But strawberry is a... <svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z" class="transition-all"></path>
      </svg>
    </button>
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform h-0 opacity-0" entertoclass="transform h-full opacity-100" leaveactiveclass="transition duration-75 ease-out" leavefromclass="transform h-full opacity-100" leavetoclass="transform h-0 opacity-0" appear="false" persisted="false" css="true">
      <!---->
    </transition-stub><button data-v-24e38f3d="" class="button-like button--primary w-full justify-between" id="headlessui-disclosure-button-v-14" aria-expanded="false" data-headlessui-state="" data-test-id="qa-button-isItFree">
      <!--v-if-->Is the app monetized? <svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z" class="transition-all"></path>
      </svg>
    </button>
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform h-0 opacity-0" entertoclass="transform h-full opacity-100" leaveactiveclass="transition duration-75 ease-out" leavefromclass="transform h-full opacity-100" leavetoclass="transform h-0 opacity-0" appear="false" persisted="false" css="true">
      <!---->
    </transition-stub><button data-v-24e38f3d="" class="button-like button--primary w-full justify-between" id="headlessui-disclosure-button-v-16" aria-expanded="false" data-headlessui-state="" data-test-id="qa-button-isMyDataSafe">
      <!--v-if-->Is my data safe? <svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z" class="transition-all"></path>
      </svg>
    </button>
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform h-0 opacity-0" entertoclass="transform h-full opacity-100" leaveactiveclass="transition duration-75 ease-out" leavefromclass="transform h-full opacity-100" leavetoclass="transform h-0 opacity-0" appear="false" persisted="false" css="true">
      <!---->
    </transition-stub><button data-v-24e38f3d="" class="button-like button--primary w-full justify-between" id="headlessui-disclosure-button-v-18" aria-expanded="false" data-headlessui-state="" data-test-id="qa-button-howToInstall">
      <!--v-if-->How to install? <svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z" class="transition-all"></path>
      </svg>
    </button>
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform h-0 opacity-0" entertoclass="transform h-full opacity-100" leaveactiveclass="transition duration-75 ease-out" leavefromclass="transform h-full opacity-100" leavetoclass="transform h-0 opacity-0" appear="false" persisted="false" css="true">
      <!---->
    </transition-stub><button data-v-24e38f3d="" class="button-like button--primary w-full justify-between" id="headlessui-disclosure-button-v-20" aria-expanded="false" data-headlessui-state="" data-test-id="qa-button-contact">
      <!--v-if-->How to get in touch? <svg style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z" class="transition-all"></path>
      </svg>
    </button>
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform h-0 opacity-0" entertoclass="transform h-full opacity-100" leaveactiveclass="transition duration-75 ease-out" leavefromclass="transform h-full opacity-100" leavetoclass="transform h-0 opacity-0" appear="false" persisted="false" css="true">
      <!---->
    </transition-stub>
  </div>
</div>
<div class="flex-container flex-col"><label class="label-like" for="reset-button">Remove all data from the app</label><button data-v-24e38f3d="" class="button-like button--danger self-end" id="reset-button" data-test-id="reset-button"><svg data-v-24e38f3d="" style="width: 1.25rem; height: 1.25rem;" class="flex-shrink-0" viewBox="0 0 24 24" aria-hidden="true">
      <path d="M9,3V4H4V6H5V19A2,2 0 0,0 7,21H17A2,2 0 0,0 19,19V6H20V4H15V3H9M9,8H11V17H9V8M13,8H15V17H13V8Z" class="transition-all"></path>
    </svg>Reset</button></div>
<build-time-stub></build-time-stub>"
`;
